pages/heroku.md

```diff
@@ -131,6 +131,115 @@ Or when using gradle:
 
 `heroku deploy:jar build/libs/*jar`
 
+## Deploying Docker to Heroku
+
+You can deploy your app as a Docker container to Heroku too. While this works, there's no Heroku setup and configuration that happens, so you have to do that manually. This documentation assumes you've already run `jhipster heroku` to deploy your app and therefore leverages the integration and add-on provisioning that this process performs.
+
+Build your Docker image:
+
+```
+./mvnw package -Pprod verify jib:dockerBuild
+```
+
+If you're using Gradle:
+
+```
+./gradlew -Pprod bootJar jibDockerBuild
+```
+
+You can test it out locally using Docker Compose.
+
+```shell
+docker-compose -f src/main/docker/app.yml up
+```
+
+Once you've confirmed everything works, create a new app on Heroku, and add it as a remote.
+
+```shell
+heroku apps:create
+git remote add docker https://git.heroku.com/<your-new-app>.git
+```
+
+Then run the commands below to deploy your JHipster app as a Docker image. Be sure to replace the `<...>` placeholders with your Heroku app name. If you don't know your app name, run `heroku apps`.
+
+```shell
+heroku container:login
+docker tag space registry.heroku.com/<heroku-app>/web
+docker push registry.heroku.com/<heroku-app>/web
+```
+
+For example:
+
+```shell
+heroku container:login
+docker tag space registry.heroku.com/fast-peak-70014/web
+docker push registry.heroku.com/fast-peak-70014/web
+```
+
+At this point, you can use the PostgreSQL and Okta add-ons you've already configured. Run the following command to get the identifiers of the add-ons from the `heroku` remote that you first deployed to.
+
+```shell
+heroku addons --remote heroku
+```
+
+Then you can attach these instances to your new application.
+
+```shell
+heroku addons:attach <postgresql-addon-name> --remote docker
+heroku addons:attach <okta-addon-name> --remote docker
+```
+
+When you use `jhipster heroku` to deploy your application, it properly configures the database for you. However, when deploying it as a Docker container, none of that happens. Therefore, you need to set a few configuration variables so your Docker container can talk to PostgreSQL. First, run the following command to get the PostgreSQL URL.
+
+```
+heroku config:get DATABASE_URL --remote docker
+```
+
+Then, set the database environment variables to match the keys that are in `application-heroku.yml`:
+
+```shell
+heroku config:set JDBC_DATABASE_URL=<value-after-@-from-last-command> --remote docker
+heroku config:set JDBC_DATABASE_USERNAME=<username-value-from-last-command> --remote docker
```

Comment:
Not exactly sure here if I understand what you mean with `username-value-from-last-command`

Reply:
The `heroku config:get DATABASE_URL` returns something like the following:

```
postgres://gesofkxyjv3umrel:87e2b57d83efe22595as2f11b7cf6195fd28f3e767e7fd39071ea092853fe4ed3@ec2-52-204-20-42.compute-1.amazonaws.com:5432/dag2k3tic441if
```

I expect that developers will know `//username:password@` syntax, but maybe I'm wrong?

Reply:
Agree. It find it a bit confusing because you use `value-after-@-from-last-command` instead of e.g. database address from last command. Maybe make it more clear by doing something like

```
heroku config:get DATABASE_URL returns a string in the form postgres://username:password@address
```
and it would be `<address>`, `<username>`. But not sure it makes it easier.

Reply:
Fixed!